docs/5-development/07-ts.md

```diff
@@ -358,6 +358,353 @@ Then, the users of your component can import the detail type and pass it to `Cus
+};
+```
+<br>
+
+**7. Do not use "any", unless absolutely necessary.**
+
+The `"any"` type, while powerful, can be **dangerous** feature as it instructs the TypeScript compiler to ignore type checking for a specific variable or expression. This can result in errors and make the code more complex to understand and maintain. Our `ESLint` usually takes care of this by enforcing best practices and avoiding its usage.
+
+<br>
+
+### TypeScript specific guidelines
+<br>
+
+**1. When to use "import type" ?**
+
+The `import` keyword is used to import values from a module, while `import type` is used to import only the type information of a module without its values. This type information can be used in type annotations and declarations.
+<br>
+
+For clarity, it is recommended to keep ***type*** and ***non-type*** imports on separate lines and explicitly mark types with the `type` keyword, like in the following example:
+
+```ts
+// This line
+
+import I18nBundle, { getI18nBundle, I18nText } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+```
+```ts
+
+// Should be split into 
+
+// Named export (function) used into the the component class
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+
+// Default export used into the the component class.
+// I18nBundle is a class constructor, but in the current example it's used 
+// as a type for a variable to which the class will be assinged.
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+
+// named type export
+import type { I18nText } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+```
+
+<br>
+
+**2. When should we use `"!"` operator in component's file ?**
+
+The `!` operator in TypeScript is used to indicate that a value is not `null` or `undefined` in situations where the type checker cannot conclude it.
+
+It is commonly used when working with the `this.getDomRef()` and `this.shadowRoot` properties in our web components. The return types of these properties, `HTMLElement | null` and `ShadowRoot | null`, respectively, are marked with `null` because there may be instances when these values are not yet available.
+
+This operator can also be used in other situations where TypeScript does not understand the framework's lifecycle, for example, when working with custom elements.
+
+In short, the `!` operator is a useful tool for ensuring that a value is not `null` or `undefined` in cases where the type checker cannot determine this on its own.
+
+For example:
+
+```ts
+import UI5Element from "sap/ui/core/Element";
+
+class Example extends UI5Element {
+	testProperty?: string;
+
+	onBeforeRendering() {
+		this.testProperty = "Some text";
+	}
+
+	onAfterRendering() {
+		// here TypeScript will complain about that the testProperty may be undefined
+		// in order of its definition and because it doesn't understand the framework's lifecycle
+		const varName: string = this.testProperty!;
+	}
+}
+```
+<br>
+
+**3. Generics in TypeScript.**
```

Comment:
Usage of Generics is a bit better to me, because Generics in TypeScript sounds like the section should explain the Generics in general as in the TS Docs, while we only give examples of several places you make use of it.